src/tools/interop/idt/capture/platform/android/android.py

```diff
@@ -0,0 +1,220 @@
+            except ValueError:
+                pass
+        valid_ip_no_port = valid_ipv4 and not port_entered
+        valid_ip_valid_port = valid_ipv4 and valid_port
+        return valid_ip_no_port or valid_ip_valid_port
+
+    def _check_connect_wireless_adb(self, temp_device_id: str) -> None:
+        # TODO: Wireless debugging does not work from container
+        if Android._is_connection_str(temp_device_id):
+            connect_command = f"adb connect {temp_device_id}"
+            print(
+                f"Detected connection string; attempting to connect: {connect_command}")
+            BashRunner(connect_command, sync=True, capture_output=False)
+            self.get_adb_devices()
+
+    def _device_id_user_input(self) -> None:
+        print('If there is no output below, press enter after connecting your phone under test OR')
+        print('Enter (copy paste) the target device id from the list of available devices below OR')
+        print('Enter $IP4:$PORT to connect wireless debugging.')
+        self._log_adb_devices()
+        temp_device_id = input('').strip()
+        self._check_connect_wireless_adb(temp_device_id)
+        if self._only_one_device_connected():
+            self._set_device_if_only_one_connected()
+        elif temp_device_id not in self.adb_devices:
+            print('Entered device not in adb devices!')
+        else:
+            self.device_id = temp_device_id
+
+    def _choose_device_id(self) -> None:
+        """
+        Prompts the user to select a single device ID for this transport
+        If only one device is ever connected, use it.
+        """
+        self._set_device_if_only_one_connected()
+        while self.device_id not in self.get_adb_devices():
+            self._device_id_user_input()
+        print(f'Selected device {self.device_id}')
+
+    def _authorize_adb(self) -> None:
+        """
+        Prompts the user until a single device is selected and adb is auth'd
+        """
+        self.get_adb_devices()
+        self._choose_device_id()
+        while not self.get_adb_devices()[self.device_id]:
+            print('Confirming authorization, press enter after auth')
+            input('')
+        print(f'Target android device ID is authorized: {self.device_id}')
+
+    def check_screen(self) -> bool:
+        screen_cmd_output = self.run_adb_command(
+            self.check_screen_command, capture_output=True)
+        return "true" in screen_cmd_output.get_captured_output()
+
+    def ensure_screen_is_on(self) -> None:
+        screen_on = self.check_screen()
+        while not screen_on:
+            input(
+                "Please turn the screen on so screen recording can start, press enter once on!")
+            screen_on = self.check_screen()
+            if not screen_on:
+                print("Screen is still not on for recording!")
+
+    def start_streaming(self) -> None:
+        if self.screen_proc is None or not self.screen_proc.command_is_running():
```

Comment:
Again, the "command is already running" checks make me feel the call flow through the framework isn't clear enough. There should be a clear top level start session that has clear lifecycle calls. Internally, it shouldn't need to check the state of previuos requests.

Reply:
Thanks, I'll fix this and above. Generally agreed with many of the  comments without responses, just going to push and resolve instead of responding to those that don't warrant discussion.

Thanks again for such thorough review.

Reply:
Thanks, have taken care of this now!